modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/translator/DeepLClient.java

```diff
@@ -0,0 +1,148 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.translation.translator.deepl.internal.translator;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.servlet.HttpHeaders;
+import com.liferay.portal.kernel.url.URLBuilder;
+import com.liferay.portal.kernel.util.ContentTypes;
+import com.liferay.portal.kernel.util.Http;
+import com.liferay.translation.translator.deepl.internal.constants.DeepLConstants;
+import com.liferay.translation.translator.deepl.internal.model.SupportedLanguage;
+import com.liferay.translation.translator.deepl.internal.model.TranslateResponse;
+import com.liferay.translation.translator.deepl.internal.util.JSONUtil;
+
+import java.io.IOException;
+
+import java.util.List;
+
+import javax.ws.rs.core.Response;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Yasuyuki Takeo
+ */
+@Component(immediate = true, service = DeepLClient.class)
+public class DeepLClient {
+
+	public TranslateResponse execute(
+			String authKey, String text, String sourceLanguageId,
+			String targetLanguageId, String url)
+		throws IOException {
+
+		return JSONUtil.toObject(
+			_fetch(authKey, text, sourceLanguageId, targetLanguageId, url),
+			TranslateResponse.class);
+	}
+
+	public List<SupportedLanguage> verifySupportedLanguage(
+			String authKey, String target, String url)
+		throws IOException {
+
+		return JSONUtil.toObject(
+			_fetchSupportedLanguage(authKey, target, url),
+			new TypeReference<List<SupportedLanguage>>() {
+			});
+	}
+
+	private String _fetch(
+			String authKey, String text, String sourceLanguageId,
+			String targetLanguageId, String url)
+		throws IOException {
+
+		Http.Options options = new Http.Options();
+
+		options.setLocation(
+			URLBuilder.create(
+				url
+			).addParameter(
+				DeepLConstants.AUTH_KEY, authKey
+			).build());
+
+		options.addHeader(
+			HttpHeaders.CONTENT_TYPE,
+			ContentTypes.APPLICATION_X_WWW_FORM_URLENCODED);
+		options.addPart(DeepLConstants.AUTH_KEY, authKey);
+		options.addPart(DeepLConstants.TEXT, text);
+		options.addPart(DeepLConstants.SOURCE_LANG, sourceLanguageId);
+		options.addPart(DeepLConstants.TARGET_LANG, targetLanguageId);
+		options.setMethod(Http.Method.POST);
+
+		Http.Response response = options.getResponse();
+
+		Response.Status status = Response.Status.fromStatusCode(
+			response.getResponseCode());
+
+		if (status == Response.Status.OK) {
+			return _http.URLtoString(options);
+		}
+		else if (status == Response.Status.TOO_MANY_REQUESTS) {
+			_log.error(
+				"Ths status is TOO_MANY_REQUESTS. Please retry after a while.");
+
+			return "";
```

Comment:
We shouldn't ignore errors like this. This should fail with a meaningful exception (create one yourself), and the error should be captured up in the call chain.